### PR TITLE
Additionally test KVStore mode in E2E/IPSec workflows

### PR DESCRIPTION
**Pull request by giorio94**
Modify a few matrix entries of the Cilium E2E Upgrade and IPSec workflows to configure Cilium in kvstore mode, to cover this functionality here as well in addition to the clustermesh workflows. In detail, the etcd instance is executed as a pod running in host network, which is setup via the `kind-kvstore-start` makefile target. The matrix entries are selected trying to cover the most common combinations, that is native-routing/tunneling, KPR off/on and wireguard off/on, and avoiding incompatible options (mainly Egress Gateway and Mutual Authentication).

### CODE REVIEW

**Comment by giorio94** [comment]
Starting to open for review to gather feedback. The first commit is from https://github.com/cilium/cilium/pull/35646 which shall be merged first (hence I marked this one as blocked).

**Comment by giorio94** [comment]
Rebased onto main to drop the commit from https://github.com/cilium/cilium/pull/35646.

File: .github/workflows/conformance-ipsec-e2e.yaml [hunk 1]
```diff
@@ -199,6 +201,19 @@ jobs:
           kind-params: "${{ steps.kind-params.outputs.params }}"
           kind-image: ${{ env.KIND_K8S_IMAGE }}
 
+      - name: Start Cilium KVStore
+        id: kvstore
+        if: matrix.kvstore == 'true'
```
Comment (sayboras) [inline .github/workflows/conformance-ipsec-e2e.yaml:206]: do we need to do ${{ }}

Comment (giorio94) [reply .github/workflows/conformance-ipsec-e2e.yaml:206]: It is not strictly necessary for simple expressions: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/using-conditions-to-control-job-execution#overview

Comment (sayboras) [reply .github/workflows/conformance-ipsec-e2e.yaml:206]: ah nice TIL, thanks.

**Review by sayboras — APPROVED** [review]

**Comment by giorio94** [comment]
Rebased onto main to fix a conflict.

**Review by brlbil — APPROVED** [review]

**Review by ldelossa — APPROVED** [review]

**Pull request merged**